### ISSUE TITLE
Design consideration for the future

### ISSUE THREAD

**Issue by tommyent** [joaomgcd/JoinChrome#28]
With the addition of features I think the design might benefit from some refactoring. For example I can choose one of the 16 options now by scrolling but using a grid layout I can do that without the scroll. This right now is the biggest on mobile devices but with things like EventGhost being added to desktop that's starting to grow. 

A lot of other exts are using a wider layout as well. Pushbullet and the new Hangouts are among the few. 

One other thing that brought this up for me was the push history. It feels like it should be in the extension itself and making it another tab up top would be nice vs having to select the device and pick from the list. So you would have 
Devices  Notifications  SMS/Phone  Push History
Push history would have the devices under it to click on those

Happy to help in anyway I can @mrkevinlin I know you're busy with school but anyway I can help let me know.

![screenshot 2016-09-15 17 35 48](https://cloud.githubusercontent.com/assets/509970/18568456/6846695a-7b6b-11e6-957d-4d3462292f8b.png)

**Comment by tommyent** [comment]
Oh one other thing at least for English is a lot of commands could be shorter 
Send current tab to your device => Send current tab
Send a Tasker Command => Send Tasker Command
Paste Clipboard on your device => Paste Clipboard
Check Device's Push History => Check Push History (this would change with above proposal)
Send an SMS message => Send SMS
Send a file to your device => Send file
etc. 
anything to do with the device is redundant since you are picking it. Also words like a, an, etc can be left out freeing up space but the user still knows what will happen

**Comment by mrkevinlin** [comment]
I like these ideas. 

Especially the one about increasing the width so that we can have a conversation list to the left of the actual conversations, removing the need for our funky "back" paradigm.

**Comment by joaomgcd** [comment]
Thank you very much for your feedback.
Just a few comments:
- When I added the EventGhost command I removed the Tasker command that was there by mistake, so no real commands were added :)
- I want to add an option to disable commands so that people don't have the ones they don't need
- How would you fit another "Push History" tab there? :) I thought about that, but saw no good way of putting it there.
- I agree that some words in the commands are redundant. Actually, if you remember, commands were a lot shorter when I first implemented them. But then that was changed to make it more "user friendly" I guess. Or rather more "beginner friendly". I guess it's less daunting for a new user to see a list of clearly written out commands then a grid of more "efficient" commands.

**Comment by tommyent** [comment]
- Touché. My point was more that as things get added it will become a bigger scroll. EventGhost was just the last one added. 😃  
- Disabling would be nice
- A wider layout would accommodate this and as @mrkevinlin pointed out fix "our funky "back" paradigm"
- I guess I can see this but at the same time consider all the apps out there that have zero text and just an icon. Or even the text area that I am typing into now. It says "Write" | "Preview" then a bunch of icons to the left. It could say "Type you comment" and "Preview your comment" and years ago it may have but I think today it goes without saying. Then there is the icons which could have Create an unordered list, Create a ordered list, etc. Either way with the ability to hide this would be less of an issue since a grid would probably not be necessary for most.

That being said I think I find the grid easier to read and use however my son disagrees and thinks the list is easier 😏  which bring me back to one of my original ideas which is having more than one theme. There is a lot that could be done with just injecting a css file.

**Comment by joaomgcd** [comment]
Not dismissing the wider layout but how about just showing the push history where the commands are (replacing the commands) and have a back arrow to show the commands again? Putting the push history in its own tab is probably overkill I guess since I feel people won't be needing it as often as for example sending SMS or checking out notifications.

Yeah, maybe have a way to toggle between grid and list would be ideal, you're right :)

**Comment by joaomgcd** [comment]
yeah, that could be an option too. But in that case wouldn't that be the same as opening a new tab? :)
